sanic/sanic.py

```diff
@@ -18,7 +18,10 @@
 
 
 class Sanic:
-    def __init__(self, name, router=None, error_handler=None):
+    def __init__(self, name=None, router=None, error_handler=None):
+        if not name:
```

Comment:
I think it would be better to use `if name is None` or `if isinstance(name, NoneType)` to account for falsey like values like `''`